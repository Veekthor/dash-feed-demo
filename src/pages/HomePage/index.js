import { HomePageContainer } from "./index.styled";
import Feed from "./modules/Feed";
import Header from "./modules/Header";
import SidePanel from "./modules/SidePanel";

const HomePage = () => {
  return (
    <>
      <Header />
      <HomePageContainer>
        <SidePanel />
        <Feed />
      </HomePageContainer>
    </>
  );
};

export default HomePage;
