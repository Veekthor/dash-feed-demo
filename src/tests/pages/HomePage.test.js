import renderer from "react-test-renderer";
import HomePage from "../../pages/HomePage";
import { loginResponse } from "../fixtures/loginResponse";
import { render, screen, waitFor } from "@testing-library/react";
import ProtectedRoute from "../../components/ProtectedRoute";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../context/authContext";
import postsResponse from "../fixtures/postsResponse";
import { act } from "react-dom/test-utils";
describe("HomePage Tests", () => {
  const localStore = {
    userDetails: JSON.stringify(loginResponse.user),
    token: process.env.TEST_ACCESS_TOKEN,
  };
  beforeAll(() => {
    global.Storage.prototype.setItem = jest.fn((key, value) => {
      localStore[key] = value;
    });
    global.Storage.prototype.getItem = jest.fn((key) => localStore[key]);
  });

  afterAll(() => {
    global.Storage.prototype.setItem.mockReset();
    global.Storage.prototype.getItem.mockReset();
    fetchMock.resetMocks();
  });

  it("should match snapshots", () => {
    const component = renderer.create(<HomePage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should fetch posts", async () => {
    fetchMock.resetMocks();
    fetchMock.mockResponseOnce(JSON.stringify(postsResponse));
    const expectedUrl =
      process.env.REACT_APP_API_BASE_URL + "/posts/feed?page=1";
    const expectedOptions = {
      headers: { "Content-Type": "application/json" },
      method: "GET",
    };
    act(() => {
      render(
        <AuthContext.Provider value={{ user: loginResponse.user }}>
          <MemoryRouter initialEntries={["/feed"]}>
            <ProtectedRoute>
              <HomePage />
            </ProtectedRoute>
          </MemoryRouter>
        </AuthContext.Provider>
      );
    });

    await waitFor(() =>
      expect(screen.queryByText("Loading...")).not.toBeInTheDocument()
    );

    expect(fetch).toHaveBeenCalledWith(expectedUrl, expectedOptions);

    const posts = screen.getAllByTestId("post-container");
    expect(posts).toHaveLength(postsResponse.posts.length);
  });
});
