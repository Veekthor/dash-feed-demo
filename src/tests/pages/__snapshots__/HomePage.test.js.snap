// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Tests should match snapshots 1`] = `
Array [
  <header
    className="sc-kAyceB iXdGGS"
  >
    <div>
      <input
        aria-label="Search"
        name="search"
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
  </header>,
  <div
    className="sc-aXZVg gBwDzS"
  >
    <nav
      className="sc-imWYAI jGrkqA"
    >
      <ul>
        <li
          className="active"
        >
          Home
        </li>
        <li>
          My Profile
        </li>
        <li>
          Connections
        </li>
        <li>
          Messages
        </li>
        <li>
          Settings
        </li>
        <li
          onClick={[Function]}
          role="button"
        >
          Logout
        </li>
      </ul>
    </nav>
    <main
      className="sc-dcJsrY ceaRFB"
    >
      <div
        className="sc-eqUAAy hIDAJa card"
      >
        <div
          className="sc-fqkvVR hOFLUZ input-container"
        >
          <img
            alt="Avatar"
            className="avatar"
          />
          <textarea
            onChange={[Function]}
            placeholder="What's happening?"
            value=""
          />
        </div>
        <div>
          <button
            className="sc-gEvEer hmIAwj"
            disabled={true}
            onClick={[Function]}
          >
            Post
          </button>
        </div>
      </div>
      <section>
        <p
          className="card"
        >
          No Results found
        </p>
      </section>
    </main>
  </div>,
]
`;
